subsys: logging: backends: fix mqtt log backend on client disconnect

rlux-bluezone opened the pull request:
The MQTT logging backend can cause the logging thread to take over idle time when the used client disconnects/fails to send the log messages. Fixed by always returning the length of the log in every case. 

Fixes zephyrproject-rtos/zephyr#101064

Signed-off-by: Roland Lux <roland.lux@blue-zone.at>

rlubos reviewed (commented):
The change looks good but why 3 commits? @rlux-bluezone  please squash them, and use `logging: backend: mqtt: ` for the commit title prefix.

rlux-bluezone committed:
logging: backend: mqtt: fix mqtt log backend on client disconnect

The MQTT logging backend can cause the logging thread
to take over idle time when the used client disconnects/fails
to send the log messages.
Fixed by always returning the length of the log in every case.

Signed-off-by: Roland Lux <roland.lux@blue-zone.at>
